src/azure-cli-core/azure/cli/core/extension/operations.py

```diff
@@ -296,6 +307,7 @@ def add_extension(cmd=None, source=None, extension_name=None, index_url=None, ye
     if extension_name:
         cmd_cli_ctx.get_progress_controller().add(message='Searching')
         ext = None
+        set_extension_management_detail(extension_name, version)
```

Comment:
`version` may be None, the information will be overwritten if the extension is resolved from index. If before that, the command exits, then the information set here can be useful.